### PR TITLE
added initial documentation for link aggregation

### PR DESCRIPTION
**Pull request by jklare**
Signed-off-by: Jan Klare <jan.klare@bisdn.de>

### CODE REVIEW

File: network_configuration/link_aggregation.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+
+## Example bonding configuration topology
+
+In the following configuration examples, we are going to use the topology shown below. It consists of two switches which are directly connected to each other on port 52 and 54. These two interfaces are then bonded using bond mode `802.3.ad` to achieve redundancy and bandwidth aggregation at the same time. The bond created is called bond2 on each side. Additionally each switch has port 7 and 8 bonded the same way into bond1 and these links are directly connected to servers which are considered source and sink for this configuration scenario. The servers themselves also use LACP to create bond1 out of their interfaces eno7 and eno8. To finally be able to send traffic between the two servers, all bonds on the switches are attached to a bridge called `swbridge` and each bond1 interface on the servers gets an IP address that should be reachable from the corresponding other server via the two switches in between.
+
+```
+ +-------------------------------------------+   +-------------------------------------------+
```
Comment (hilmarm) [inline network_configuration/link_aggregation.md:20]: We could use more generic terms like 'port_A', but also not really necessary imo.

File: network_configuration/link_aggregation.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+To configure the scenario described and shown above, we first create the two bond devices themselves on both switches:
+
+```
+ip link add bond1 type bond
+ip link add bond2 type bond
+```
+
+After the bonds are created, we set the mode to 802.3ad (LACP):
+```
+ip link set bond1 type bond mode 802.3ad
```
Comment (hilmarm) [inline network_configuration/link_aggregation.md:62]: It is possible to combine the creation of the bond and setting the bond mode at the same time. i.e.
``ip l add bond2 type bond mode 802.3ad``
Also not necessary. It is worth mentinoing that Linux defaults to 'load balancing (round-robin)'

File: network_configuration/link_aggregation.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+ |    +------------+            |                             |              +------------+  |
+ |       10.0.3.1/24            |                             |                 10.0.3.2/24  |
+ |                              |                             |                              |
+ |                     server-1 |                             |                     server-2 |
+ +------------------------------+                             +------------------------------+
+```
+
+## Bonding configuration with iproute2
+
+To configure the scenario described and shown above, we first create the two bond devices themselves on both switches:
```
Comment (hilmarm) [inline network_configuration/link_aggregation.md:53]: All commands have to be run with root privileges

**Review by rr-santos — COMMENTED** [review]
See comments regarding introductory text. Will test the configuration instructions and give feedback on that, but the text about that seems very clear.

File: network_configuration/link_aggregation.md (hunk 1 above)
Comment (jklare) [reply network_configuration/link_aggregation.md:20]: i think beeing specific about the names of the ports is easier to understand

File: network_configuration/link_aggregation.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+In this section, we provide examples on how to configure bonding using [iproute2](#bonding-configuration-with-iproute2) and [systemd-networkd](#bonding-configuration-with-systemd-networkd) networks in BISDN Linux.
```
[the remark this answers is not in the document]
Comment (jklare) [reply network_configuration/link_aggregation.md:19]: currently we do not have this section and have to also check for each ASIC which modes will be supported

File: network_configuration/link_aggregation.md (hunk 2 above)
Comment (jklare) [reply network_configuration/link_aggregation.md:75]: i think we should point people to the official docs here, but i will also add this

File: network_configuration/link_aggregation.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+ip link add bond2 type bond
+```
+
+After the bonds are created, we set the mode to 802.3ad (LACP):
+```
+ip link set bond1 type bond mode 802.3ad
+ip link set bond2 type bond mode 802.3ad
+```
+
+To enslave the links to their corresponding bonds, we first set them down and then update their configuration my setting their master to the bond we created before:
```
Comment (hilmarm) [inline network_configuration/link_aggregation.md:79]: configuration _my_ setting > configuration **by** setting

**Comment by hilmarm** [comment]
I fixed the spelling mistakes in order to merge this for the release @jklare .

**Pull request merged**